packages/tx/src/baseTransaction.ts

```diff
@@ -285,7 +292,7 @@ export abstract class BaseTransaction<TransactionObject> {
     }
 
     const msgHash = this.getMessageToSign(true)
-    const { v, r, s } = ecsign(msgHash, privateKey)
+    const { v, r, s } = sign(msgHash)
```

Comment:
We would need to await here.

Reply:
I only did it for `tx` and `util`. Maybe also for `common`. I wasn't able to build the whole thing.

Reply:
Is it a good idea to make `sign` async? This function may be used in places we don't want to refactor.

Reply:
> Is it a good idea to make sign async?

probably not, let's go with your version

Reply:
In theory it should be possible for `sign` to return a `TransactionObject | Promise<TransactionObject>`. That was what I was exploring. Since one knows what was supplied, one knows when to await. In my case TS seemed to know better...